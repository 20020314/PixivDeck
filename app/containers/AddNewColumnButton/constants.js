// @flow
import type { ADD_USER_ILLUST_TYPE } from './actionTypes.js'

export const ADD_USER_ILLUST: ADD_USER_ILLUST_TYPE =
  'AddNewColumnButton/ADD_USER_ILLUST'
