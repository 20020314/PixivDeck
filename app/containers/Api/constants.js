// @flow
import type { API_REQUEST_SUCCESS_TYPE } from './actionTypes.js'

export const API_REQUEST_SUCCESS: API_REQUEST_SUCCESS_TYPE =
  'Api/API_REQUEST_SUCCESS'
