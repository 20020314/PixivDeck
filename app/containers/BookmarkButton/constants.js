// @flow
import type {
  ADD_BOOKMARK_REQUEST_TYPE,
  ADD_BOOKMARK_SUCCESS_TYPE,
  ADD_BOOKMARK_FAILER_TYPE,
} from './actionTypes.js'

export const ADD_BOOKMARK_REQUEST: ADD_BOOKMARK_REQUEST_TYPE =
  'BookmarkButton/ADD_BOOKMARK_REQUEST'
export const ADD_BOOKMARK_SUCCESS: ADD_BOOKMARK_SUCCESS_TYPE =
  'BookmarkButton/ADD_BOOKMARK_SUCCESS'
export const ADD_BOOKMARK_FAILER: ADD_BOOKMARK_FAILER_TYPE =
  'BookmarkButton/ADD_BOOKMARK_FAILER'
