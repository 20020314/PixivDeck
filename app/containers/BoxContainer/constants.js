// @flow
import type { OPEN_PIXIV_TYPE } from './actionTypes.js'

export const OPEN_PIXIV: OPEN_PIXIV_TYPE = 'BoxContainer/OPEN_PIXIV'
