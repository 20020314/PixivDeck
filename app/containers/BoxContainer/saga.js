// @flow
import { shell } from 'electron'
import { Actions } from './actionTypes'
import { call, takeEvery, type IOEffect } from 'redux-saga/effects'

function* openPixiv({ id }: { id: number }): Generator<IOEffect, void, *> {
  yield call(
    shell.openExternal,
    `http://www.pixiv.net/member_illust.php?mode=medium&illust_id=${id}`
  )
}

export default function* root(): Generator<IOEffect, void, *> {
  yield takeEvery(Actions.OPEN_PIXIV, openPixiv)
}
