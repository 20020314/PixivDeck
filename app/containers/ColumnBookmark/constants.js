// @flow
import type {
  ADD_COLUMN_TYPE,
  ADD_COLUMN_SUCCESS_TYPE,
  SET_NEXT_URL_TYPE,
  FETCH_BOOKMARK_TYPE,
  FETCH_BOOKMARK_SUCCESS_TYPE,
  FETCH_BOOKMARK_FAILRE_TYPE,
  FETCH_NEXT_BOOKMARK_TYPE,
  FETCH_NEXT_BOOKMARK_SUCCESS_TYPE,
  FETCH_NEXT_BOOKMARK_FAILRE_TYPE,
} from './actionTypes.js'

export const ADD_COLUMN: ADD_COLUMN_TYPE = 'ColumnBookmark/ADD_COLUMN'
export const ADD_COLUMN_SUCCESS: ADD_COLUMN_SUCCESS_TYPE =
  'ColumnBookmark/ADD_COLUMN_SUCCESS'
export const SET_NEXT_URL: SET_NEXT_URL_TYPE = 'ColumnBookmark/SET_NEXT_URL'
export const FETCH_BOOKMARK: FETCH_BOOKMARK_TYPE =
  'ColumnBookmark/FETCH_BOOKMARK'
export const FETCH_BOOKMARK_SUCCESS: FETCH_BOOKMARK_SUCCESS_TYPE =
  'ColumnBookmark/FETCH_BOOKMARK_SUCCESS'
export const FETCH_BOOKMARK_FAILRE: FETCH_BOOKMARK_FAILRE_TYPE =
  'ColumnBookmark/FETCH_BOOKMARK_FAILRE'
export const FETCH_NEXT_BOOKMARK: FETCH_NEXT_BOOKMARK_TYPE =
  'ColumnBookmark/FETCH_NEXT_BOOKMARK'
export const FETCH_NEXT_BOOKMARK_SUCCESS: FETCH_NEXT_BOOKMARK_SUCCESS_TYPE =
  'ColumnBookmark/FETCH_NEXT_BOOKMARK_SUCCESS'
export const FETCH_NEXT_BOOKMARK_FAILRE: FETCH_NEXT_BOOKMARK_FAILRE_TYPE =
  'ColumnBookmark/FETCH_NEXT_BOOKMARK_FAILRE'
