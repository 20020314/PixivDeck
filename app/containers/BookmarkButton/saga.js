// @flow
import { call, put, takeEvery, type IOEffect } from 'redux-saga/effects'
import { get } from '../Api/sagas'
import * as api from 'containers/Api/sagas'
import * as Actions from './constants'
import * as actions from './actions'
import type { Restrict } from './types'

type Props = {
  id: number,
  restrict: Restrict,
}

export function* bookmark({ id, restrict }: Props): Generator<*, void, void> {
  try {
    yield call(api.post, '/v2/illust/bookmark/add', { illustId: id, restrict })

    yield put(actions.addBookmarkSuccess(id, restrict))
  } catch (err) {
    yield put(actions.addBookmarkFailer(id, err))
  }
}

export function* deleteTask({ id }: Props): Generator<*, void, void> {
  try {
    yield call(api.post, '/v1/illust/bookmark/delete', { illustId: id })
    yield put(actions.deleteBookmarkSuccess(id))
  } catch (err) {
    yield put(actions.deleteBookmarkFailer(id, err))
  }
}

function* success({ id }: Props) {
  try {
    yield call(get, `/v1/illust/detail?illust_id=${id}`, true)
  } catch (err) {}
}

function* root(): Generator<IOEffect, void, *> {
  yield takeEvery(Actions.ADD_BOOKMARK_REQUEST, bookmark)
  yield takeEvery(Actions.DELETE_BOOKMARK_REQUEST, deleteTask)
  yield takeEvery(
    [Actions.ADD_BOOKMARK_SUCCESS, Actions.DELETE_BOOKMARK_SUCCESS],
    success
  )
}

export default root
