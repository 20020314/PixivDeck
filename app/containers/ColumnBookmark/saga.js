// @flow
import union from 'lodash/union'
import { addColumn } from 'containers/ColumnManager/actions'
import { makeSelectInfo } from 'containers/LoginModal/selectors'
import { getToken } from 'containers/LoginModal/saga'
import { getRequest, fetchAuth } from '../../api/client'
import * as Actions from './constants'
import * as actions from './actions'
import type { ColumnId } from './reducer'
import { makeSelectColumn, makeSelectIds } from './selectors'
import { put, select, call, takeEvery } from 'redux-saga/effects'

function* addBookmarkColumn({ id }: { id: ColumnId }) {
  const ids: Array<?ColumnId> = yield select(makeSelectIds())
  if (ids.every(v => v !== id)) {
    yield put(actions.addBookmarkColumnSuccess(id))
  }

  yield put(addColumn(`bookmark-${id}`, { columnId: id, type: 'BOOKMARK' }))
}

type Props = { id: ColumnId }

function* fetchBookmark(props: Props) {
  const { id } = props
  const { illustIds } = yield select(makeSelectColumn(), props)

  try {
    const info = yield select(makeSelectInfo())
    // TODO
    const { accessToken, user: { id: userId } } = yield call(fetchAuth, info)

    const response = yield call(
      getRequest,
      `/v1/user/bookmarks/illust?user_id=${userId}&restrict=${id}`,
      null,
      accessToken
    )
    const { result } = response

    yield put(actions.setNextUrl(id, result.nextUrl))

    const nextIds = union(result.illusts, illustIds)
    yield put(actions.fetchBookmarkSuccess(id, response, nextIds))
  } catch (err) {
    yield put(actions.fetchBookmarkFailre(id))
  }
}

function* fetchNextBookmark(props: Props) {
  const { id } = props
  const { illustIds, nextUrl } = yield select(makeSelectColumn(), props)

  try {
    if (!nextUrl) {
      return
    }

    const accessToken = yield call(getToken)

    const response = yield call(getRequest, nextUrl, null, accessToken)
    const { result } = response

    yield put(actions.setNextUrl(id, result.nextUrl))

    const nextIds = union(illustIds, result.illusts)
    yield put(actions.fetchBookmarkSuccess(id, response, nextIds))
  } catch (err) {
    yield put(actions.fetchNextBookmarkFailre(id))
  }
}

export default function* root(): Generator<*, void, void> {
  yield takeEvery(Actions.ADD_BOOKMARK_COLUMN, addBookmarkColumn)
  yield takeEvery(Actions.FETCH_BOOKMARK, fetchBookmark)
  yield takeEvery(Actions.FETCH_NEXT_BOOKMARK, fetchNextBookmark)
}
