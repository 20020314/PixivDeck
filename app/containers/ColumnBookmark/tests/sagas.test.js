// @flow
import { takeEvery } from 'redux-saga/effects'
import * as sagas from '../saga'
import * as constants from '../constants'

test('root', () => {
  const gen = sagas.default()
  const next = gen.next()
  expect(next.value).toEqual(takeEvery(constants.ADD_COLUMN, sagas.addColumn))

  // next = gen.next()
  // expect(next.value).toEqual(
  //   takeEvery(constants.FETCH, sagas.fetchUserIllust)
  // )
  //
  // next = gen.next()
  // expect(next.value).toEqual(
  //   takeEvery(constants.FETCH_NEXT, sagas.fetchNextUserIllust)
  // )
})

test('fetch next', () => {
  const gen = sagas.fetchBookmark({ id: 'public', type: constants.FETCH })
  const next = gen.next()
  expect(next.value).toMatchSnapshot()
  expect(gen.next({ nextUrl: 'fake' }).value).toMatchSnapshot()
})

test('fetch first', () => {
  const gen = sagas.fetchBookmark({ id: 'public', type: constants.FETCH })
  let next = gen.next()

  expect(next.value).toHaveProperty('SELECT')
  expect(next.value).toMatchSnapshot()

  next = gen.next({ id: '1' })
  expect(next.value).toHaveProperty('SELECT')
  expect(next.value).toMatchSnapshot()

  next = gen.next({ ids: [1, 2, 3] })
  expect(next.value).toHaveProperty('CALL')
  expect(next.value).toMatchSnapshot()
})
