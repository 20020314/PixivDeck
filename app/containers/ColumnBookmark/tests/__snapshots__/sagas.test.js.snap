// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch first 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [
      Object {
        "id": "public",
      },
    ],
    "selector": [Function],
  },
}
`;

exports[`fetch first 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`fetch first 3`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/v1/user/bookmarks/illust?user_id=[object Object]&restrict=public",
      "public",
      Object {
        "addColumn": [Function],
        "addColumnSuccess": [Function],
        "fetch": [Function],
        "fetchFailre": [Function],
        "fetchNext": [Function],
        "fetchNextFailre": [Function],
        "fetchNextSuccess": [Function],
        "fetchSuccess": [Function],
        "setNextUrl": [Function],
      },
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`fetch next 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [
      Object {
        "id": "public",
      },
    ],
    "selector": [Function],
  },
}
`;

exports[`fetch next 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "fake",
      "public",
      Object {
        "addColumn": [Function],
        "addColumnSuccess": [Function],
        "fetch": [Function],
        "fetchFailre": [Function],
        "fetchNext": [Function],
        "fetchNextFailre": [Function],
        "fetchNextSuccess": [Function],
        "fetchSuccess": [Function],
        "setNextUrl": [Function],
      },
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`new 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [
      Object {
        "id": "public",
      },
    ],
    "selector": [Function],
  },
}
`;

exports[`new 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`new 3`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      Object {
        "endpoint": "/v1/user/bookmarks/illust?user_id=myid&restrict=public",
        "id": "public",
        "ids": Array [
          1,
          2,
          3,
        ],
        "order": true,
      },
      Object {
        "addColumn": [Function],
        "addColumnSuccess": [Function],
        "fetch": [Function],
        "fetchFailre": [Function],
        "fetchNext": [Function],
        "fetchNextFailre": [Function],
        "fetchNextSuccess": [Function],
        "fetchSuccess": [Function],
        "setNextUrl": [Function],
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
